rollingpin/hostsources/autoscaler.py

```diff
@@ -14,46 +11,7 @@
 
 from ..config import Option
 from ..hostsources import Host, HostSource, HostSourceError
-from ..utils import sorted_nicely
-
-
-MAX_PARALLELISM = 50
-
-
-@inlineCallbacks
-def parallel_map(iterable, fn, *args, **kwargs):
-    deferreds = []
-    parallelism_limiter = DeferredSemaphore(MAX_PARALLELISM)
-    for item in iterable:
-        d = parallelism_limiter.run(fn, item, *args, **kwargs)
-        deferreds.append(d)
-    results = yield gatherResults(deferreds)
-    returnValue(results)
-
-
-def _distribute_into(master, additions):
-    assert len(master) >= len(additions)
-
-    spread = int(math.ceil(float(len(master)) / len(additions)))
-
-    for i, item in enumerate(additions):
-        master.insert(i * spread, item)
-
-
-def interleaved(by_pool):
-    """Merge lists such that items from the same sublist are maximally apart.
-
-    This ensures that no pool gets a bunch of its servers taken down all at
-    the same time due to an unlucky host ordering.
-
-    """
-    pools_by_size = sorted(
-        by_pool.items(), key=lambda t: len(t[1]), reverse=True)
```

Comment:
Or does items return callbacks that are processed later or something fancy like that?